lib/src/base-styles/_nx-tables.scss

```diff
@@ -365,29 +365,89 @@
   }
 }
 
-.nx-cell__sort-btn {
-  height: nx-table-variables.$table-cell-header-height;
+.nx-cell {
+  .nx-cell__sort-btn {
+    height: nx-table-variables.$table-cell-header-height;
+    position: relative;
+  
+    > * {
+      position: relative;
+    }
+  
+    &::before {
+      box-sizing: border-box;
+      border-radius: 4px;
+      content: '';
+      // offset by 1px on top and 2px on the bottom
+      height: calc(100% - 3px);
+      position: absolute;
+      left: 0;
+      top: 1px;
+      width: calc(100% - 1px);
+    }
+  
+    &:hover {
+      &::before {
+        background-color: var(--nx-color-table-header-button-background-hover);
+      }
+    }
+    &:focus {
+      &::before {
+        border: var(--nx-focus-outline-width) solid var(--nx-color-interactive-border-focus);
+      }
+    }
+    &:active {
+      &::before {
+        background-color: var(--nx-color-table-header-button-background-active);
+        border: none;
+      }
+    }
+  }
+  &:first-child {
+    .nx-cell__sort-btn {
+      &::before {
+        left: 1px;
+      }
+    }
+  }
+
+  &:not(:first-child), &:not(:last-child) {
+    .nx-cell__sort-btn {
+      &::before {
+        width: 100%;
+      }
+    }
+  }
 }
 
 // adjust button sizing for ends of rows, where padding is different and there are borders
 .nx-cell {
   $edge-spacing: calc(var(--nx-spacing-4x) - 1px);
 
   &:first-child {
-    > .nx-cell__sort-btn, > .nx-cell__row-btn {
+    > .nx-cell__row-btn {
       @include btn-cell-sizing($edge-spacing, var(--nx-spacing-3x));
     }
+    > .nx-cell__sort-btn {
```

Comment:
I wasn't sure about separating `.nx-cell__sort-btn` out of this `.nx-cell` code block  as they no longer use `$edgespacing`. I thought for clarity sake it made sense to leave them with other button sizing adjustments for ends of rows.